File: AndroidVideoView/app/src/main/java/com/bowoon/android/android_videoview/RecyclerAdapter.java
package com.bowoon.android.android_videoview;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.media.ThumbnailUtils;
import android.provider.MediaStore;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.List;

interface ItemClickListener {
    void onItemClick(int position);
}

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> implements ItemClickListener {
    private Context context;
    private List<Item> items;

    public RecyclerAdapter(Context context, List<Item> items) {
        this.context = context;
        this.items = items;
    }

    public String getItemTitle(int position) {
        return items.get(position).getTitle();
    }

    @Override
    public void onItemClick(int position) {
        Toast.makeText(context, getItemTitle(position), Toast.LENGTH_SHORT).show();
        Intent intent = new Intent(context, VideoPlayerActivity.class);
        intent.putExtra("path", items.get(position).getPath());
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        context.startActivity(intent);
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        RecyclerView.ViewHolder holder = null;

        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);

        holder = new ItemViewHolder(v, this);

        return holder;
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        if (holder == null) {
            return;
        }

        ((ItemViewHolder) holder).getTitle().setText(items.get(position).getTitle());
        Bitmap thumbnail = ThumbnailUtils.createVideoThumbnail(items.get(position).getPath(), MediaStore.Images.Thumbnails.MINI_KIND);
        ((ItemViewHolder) holder).getThumbnail().setImageBitmap(thumbnail);
    }

    @Override
    public int getItemCount() {
        return this.items.size();
    }

    static class ItemViewHolder extends RecyclerView.ViewHolder {
        private TextView title;
        private ImageView thumbnail;

        public ItemViewHolder(View itemView, final ItemClickListener itemClickListener) {
            super(itemView);

            this.title = (TextView) itemView.findViewById(R.id.video_title);
            this.thumbnail = (ImageView) itemView.findViewById(R.id.video_thumbnail);

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    itemClickListener.onItemClick(getAdapterPosition());
                }
            });
        }

        public TextView getTitle() {
            return title;
        }

        public void setTitle(TextView title) {
            this.title = title;
        }

        public ImageView getThumbnail() {
            return thumbnail;
        }

        public void setThumbnail(ImageView thumbnail) {
            this.thumbnail = thumbnail;
        }
    }
}

File: AndroidVideoView/app/src/main/java/com/bowoon/android/android_videoview/adapter/RecyclerAdapter.kt
package com.bowoon.android.android_videoview.adapter

import android.media.ThumbnailUtils
import android.provider.MediaStore
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.bowoon.android.android_videoview.R
import com.bowoon.android.android_videoview.databinding.ItemLayoutBinding
import com.bowoon.android.android_videoview.vo.Item

class RecyclerAdapter(val listener: ItemClickListener) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    private var items: List<Item>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val binding = DataBindingUtil.inflate<ItemLayoutBinding>(
                LayoutInflater.from(parent.context),
                R.layout.item_layout,
                parent,
                false
        )
        return ItemViewHolder(binding)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is ItemViewHolder) {
            ThumbnailUtils.createVideoThumbnail(items!![position].path, MediaStore.Images.Thumbnails.MINI_KIND).let {
                holder.binding.videoThumbnail.setImageBitmap(it)
            }
            holder.binding.videoTitle.text = items!![position].title
            holder.binding.item = items!![position]
            holder.binding.listener = listener
        }
    }

    override fun getItemCount(): Int {
        return if (items == null) {
            0
        } else {
            items!!.size
        }
    }

    fun setItems(items: ArrayList<Item>) {
        this.items = items
        notifyDataSetChanged()
    }

    companion object ViewHolder {
        class ItemViewHolder(var binding: ItemLayoutBinding) : RecyclerView.ViewHolder(binding.root)
    }
}
